### PR TITLE
fix: scripts in package json at monorepo creation

### LINKED ISSUE #721: [Bug]: Package json scripts at monorepo root level
### Package name

core

### Package version

9.1.0

### Reproduction steps

run o3r create to create a monorepo skelethon files

`npm run '@o3r' create my-wk-name`

### Current result

This will create monorepo skeleton files and preperare o3r generators inside the monorepo.
The _package.json_ which is created contains the scripts generated by angular generator for an app which does not work with multiple pacakges.

### Expected result
The `scripts` to be empty (at the time being).

### PR DESCRIPTION
**Pull request by mrednic-1A**
## Related issues

- :bug: Fixes #721

### CODE REVIEW

File: packages/@o3r/create/src/index.ts [hunk 1]
```diff
@@ -119,8 +120,26 @@ const addOtterCore = (relativeDirectory = '.') => {
   }
 };
 
+/**
+ * When the monorepo is created, the package json will keep only the ng script
+ * @param relativeDirectory package.json directory
+ */
+const cleanPackageJsonScripts = (relativeDirectory = '.') => {
```
Comment (kpanot) [inline packages/@o3r/create/src/index.ts:127]: Should avoid as much as possible to do content change in this script.
I would suggest to do the change of the `package.json` (maybe by removing the specific scripts that does not make explicitly sense, not replacing the whole `script` field) in the `ng add @or3/core`

File: packages/@o3r/create/src/index.ts [hunk 2]
```diff
@@ -119,8 +120,26 @@ const addOtterCore = (relativeDirectory = '.') => {
...
+  const packageJsonPath = resolve(process.cwd(), relativeDirectory, 'package.json');
+  try {
+    const packageJsonContent = JSON.parse(readFileSync(packageJsonPath, { encoding: 'utf-8' })) as PackageJson;
+    packageJsonContent.scripts = { ng: 'ng' };
+    writeFileSync(packageJsonPath, JSON.stringify(packageJsonContent, null, 2));
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(error);
+    process.exit(2);
```
Comment (matthieu-crouzet) [inline packages/@o3r/create/src/index.ts:136]:
```suggestion
    process.exit(3);
```
2 is already use line 119

File: packages/@o3r/create/src/index.ts (hunk 1 above)
Comment (mrednic-1A) [reply packages/@o3r/create/src/index.ts:127]: I agree that it is not a good idea to complexify the `create`. After discussing the solution you proposed to remove the specific scripts it's what we need. I'll do the updates accordingly

File: packages/@o3r/create/src/index.ts (hunk 2 above)
Comment (mrednic-1A) [reply packages/@o3r/create/src/index.ts]: the modification on this file was fully discarded

**Review by matthieu-crouzet — APPROVED** [review]

File: packages/@o3r/core/schematics/rule-factories/monorepo/index.ts [hunk 3]
```diff
@@ -49,3 +48,31 @@ export function addWorkspacesToProject(directories: WorkspaceLayout = DEFAULT_RO
...
+ */
+export function filterPackageJsonScripts(tree: Tree, _context: SchematicContext) {
+  const rootPackageJsonPath = '/package.json';
+  if (!tree.exists(rootPackageJsonPath)) {
+    throw new SchematicsException('Root package.json does not exist');
+  }
+
+  const rootPackageJsonObject = tree.readJson(rootPackageJsonPath) as PackageJson;
+  rootPackageJsonObject.scripts ||= {};
+  rootPackageJsonObject.scripts = Object.entries(rootPackageJsonObject.scripts).reduce<Record<string, string | undefined>>((acc, [scriptName, scriptValue]) => {
```
Comment (kpanot) [inline packages/@o3r/core/schematics/rule-factories/monorepo/index.ts:65]: would be simpler with a map:
```typescript
const defaultScriptToRemote = {
  build: 'ng build',
  test: 'ng test',
  start: 'ng serve',
  watch: 'ng build --watch --configuration development'
};

rootPackageJsonObject.scripts = Object.fromEntries(
  Object.entries(rootPackageJsonObject.scripts)
    .filter(([key, value]) => value !== defaultScriptToRemote[key])
);
```

**Review by fpaul-1A — APPROVED** [review]

**Review by kpanot — APPROVED** [review]

**Pull request merged**